File: src/core/tool_context.ts
/**
 * Tool Context — 全局工具上下文
 * 
 * 允许工具函数访问 Agent 级别的功能（如用户交互、子代理生成），
 * 而不需要修改 ToolFn 签名。
 * 
 * 与 Python tool_context.py 对应
 */

export interface ToolContext {
  /** 向用户提问并获取回答（交互模式） */
  askUser?: (question: string) => Promise<string>;
  /** 生成子代理执行独立任务 */
  spawnSubagent?: (task: string, model?: string) => Promise<string>;
  /** 当前工作目录 */
  workDir?: string;
  /** 是否为非交互模式（管道/CI） */
  nonInteractive?: boolean;
  /** 当前 Agent 配置（只读引用） */
  agentConfig?: Record<string, unknown>;
}

const _ctx: ToolContext = {};

export function setToolContext(ctx: Partial<ToolContext>): void {
  Object.assign(_ctx, ctx);
}

export function getToolContext(): ToolContext {
  return _ctx;
}

export function clearToolContext(): void {
  for (const k of Object.keys(_ctx)) {
    delete (_ctx as Record<string, unknown>)[k];
  }
}

File: python/cortex_agent/tool_context.py
"""
Tool Context — 全局工具上下文

允许工具函数访问 Agent 级别的功能（如用户交互、子代理生成），
而不需要修改工具函数签名。

与 TS tool_context.ts 对应
"""

from typing import Optional, Callable, Any, Dict

_ctx: Dict[str, Any] = {}


def set_tool_context(ctx: Dict[str, Any]) -> None:
    """设置工具上下文"""
    _ctx.update(ctx)


def get_tool_context() -> Dict[str, Any]:
    """获取工具上下文"""
    return _ctx


def clear_tool_context() -> None:
    """清空工具上下文"""
    _ctx.clear()
